qiskit_machine_learning/circuit/library/qnn_circuit.py

```diff
@@ -0,0 +1,243 @@
+    with the corresponding number of qubits is used.
+
+    At least one parameter has to be provided. If a feature map and an ansatz is provided, the
+    number of qubits must be the same.
+
+    In case number of qubits is provided along with either a feature map, an ansatz or both, a
+    potential mismatch between the three inputs with respect to the number of qubits is
+    resolved by constructing the ``QNNCircuit`` with the given number of qubits.
+    If one of the ``QNNCircuit`` properties is set after the class construction, the circuit is
+    is adjusted  to incorporate the changes. This is, a new valid configuration that considers the
+    latest property update will be derived. This ensures that the classes properties are consistent
+    at all times.
+
+    Example:
+
+    .. code-block:: python
+
+        from qiskit_machine_learning.circuit.library import QNNCircuit
+        qnn_qc = QNNCircuit(2)
+        print(qnn_qc)
+        # prints:
+        #      ┌──────────────────────────┐»
+        # q_0: ┤0                         ├»
+        #      │  ZZFeatureMap(x[0],x[1]) │»
+        # q_1: ┤1                         ├»
+        #      └──────────────────────────┘»
+        # «     ┌──────────────────────────────────────────────────────────┐
+        # «q_0: ┤0                                                         ├
+        # «     │  RealAmplitudes(θ[0],θ[1],θ[2],θ[3],θ[4],θ[5],θ[6],θ[7]) │
+        # «q_1: ┤1                                                         ├
+        # «     └──────────────────────────────────────────────────────────┘
+
+        print(qnn_qc.num_qubits)
+        # prints: 2
+
+        print(qnn_qc.input_parameters)
+        # prints: ParameterView([ParameterVectorElement(x[0]), ParameterVectorElement(x[1])])
+
+        print(qnn_qc.weight_parameters)
+        # prints: ParameterView([ParameterVectorElement(θ[0]), ParameterVectorElement(θ[1]),
+        #         ParameterVectorElement(θ[2]), ParameterVectorElement(θ[3]),
+        #         ParameterVectorElement(θ[4]), ParameterVectorElement(θ[5]),
+        #         ParameterVectorElement(θ[6]), ParameterVectorElement(θ[7])])
+    """
+
+    def __init__(
+        self,
+        num_qubits: int | None = None,
+        feature_map: QuantumCircuit | None = None,
+        ansatz: QuantumCircuit | None = None,
+    ) -> None:
+        """
+        Although all parameters default to None at least one parameter must be provided, to determine
+        the number of qubits from it, when the instance is created.
+
+        If more than one parameter is passed:
+        1) If num_qubits is provided the feature map and/or ansatz supplied will be overridden to
+        circuits with num_qubits, as long as the respective circuit supports updating its number of
+        qubits.
+        2) If num_qubits is not provided the feature_map and ansatz must be set to the same number
+        of qubits.
```

Comment:
![image](https://user-images.githubusercontent.com/40241007/229901284-c153740f-377f-4883-85e2-26aafcba6c6e.png)

To get a list you need a blank line preceding it. Otherwise it all gets wrapped into a paragraph as you can see.